### PR TITLE
Virtual keyboard labels v2

### PR DESCRIPTION
**Pull request by wilckerson**
Adjusting the Virtual Keyboard to display notes info into V2.
Note: It was not possible to display labels because it looks like the V2 doesn't have this feature.
 
image
image

### CODE REVIEW

**Comment by wilckerson** [comment]
The state from v2 doesn't same the same pros as v3, I will need to finish the adjustments later.

**Comment by wilckerson** [comment]
@frostburn Would you mind reviewing this v2 backport? Thanks

File: src/components/VirtualKeyInfo.vue [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<script setup lang="ts">
+import { formatHertz, formatExponential, formatCents } from '@/utils'
+
+const props = defineProps<{
+  cent: number
```
Comment (frostburn) [inline src/components/VirtualKeyInfo.vue:5]: Should be `cents` instead. A single cent only refers to a ratio of 2^(1/1200).
Missed this on v3, so I'll fix it there myself.

File: src/components/VirtualKeyInfo.vue [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ratio: number
+  frequency: number
+  showCent: boolean
```
Comment (frostburn) [inline src/components/VirtualKeyInfo.vue:8]: See above.

File: src/stores/state.ts [hunk 3]
```diff
@@ -58,6 +58,11 @@ export const useStateStore = defineStore('state', () => {
   const decimalFractionDigits = ref(parseInt(storage.getItem('decimalFractionDigits') ?? '5', 10))
   const showVirtualQwerty = ref(storage.getItem('showVirtualQwerty') === 'true')
   const midiOctaveOffset = ref(parseInt(storage.getItem('midiOctaveOffset') ?? '-1', 10))
+  const showKeyboardCents = ref(storage.getItem('showKeyboardCents') !== 'false')
```
Comment (frostburn) [inline src/stores/state.ts:61]: These new settings are missing the associated watchers. Might make sense to backport `syncValues`.

**Comment by frostburn** [comment]
SW2 doesn't have user-defined labels, but it does have names. See how `TuningTable.vue` populates the name field of `TuningTableRow.vue`.

**Comment by frostburn** [comment]
Looking good in general. :+1:

File: src/stores/state.ts [hunk 4]
```diff
@@ -70,6 +76,24 @@ export const useStateStore = defineStore('state', () => {
...
+    showKeyboardCents,
+    showKeyboardRatio,
+    showKeyboardFrequency,
+    intervalMatrixIndexing,
+    deactivationCode,
+    equaveUpCode,
+    equaveDownCode,
+    degreeUpCode,
+    degreeDownCode
+  })
```
Comment (wilckerson) [inline src/stores/state.ts:95]: I have backported this as requested.

File: src/components/VirtualKeyInfo.vue [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  showCents: boolean
+  showRatio: boolean
+  showFrequency: boolean
+}>()
+</script>
+
+<template>
+  <div class="key-info">
+    <div v-if="props.showLabel">
+      <strong>{{ props.label }}</strong>
```
Comment (wilckerson) [inline src/components/VirtualKeyInfo.vue:19]: Displaying label same as the `TuningTable.vue`

File: src/components/VirtualKeyInfo.vue [hunk 6]
```diff
@@ -0,0 +1,46 @@
+<script setup lang="ts">
+import { formatHertz, formatExponential, formatCents } from '@/utils'
+
+const props = defineProps<{
+  label:string
+  cents: number
+  ratio: number
+  frequency: number
+  showLabel: boolean
+  showCents: boolean
```
Comment (wilckerson) [inline src/components/VirtualKeyInfo.vue:10]: Fixed the  plural of `cents`

**Comment by wilckerson** [comment]
@frostburn I have added the required adjustments. Thanks

File: src/stores/state.ts (hunk 4 above)
Comment (frostburn) [reply src/stores/state.ts:95]: Now it double-watches the existing values, though.

**Review by frostburn — APPROVED** [review]
The code looks clean and the UI works similar to v3. The issue with double-watching isn't fatal and can easily be fixed after merging.

**Pull request merged**